frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/pvc-upload/pvc-upload.ts

```diff
@@ -0,0 +1,138 @@
+      },
+    });
+    cy.deleteResource({
+      kind: 'VirtualMachine',
+      metadata: {
+        name: vmData.name,
+        namespace: vmData.namespace,
+      },
+    });
+    cy.exec('rm -fr /tmp/cirror*');
+  });
+
+  describe('test pvc upload via form', () => {
+    it('ID(CNV-4778) No warning message shows when image format is supported', () => {
+      pvc.form.open();
+      for (const img of imageFormats) {
+        cy.exec(`touch ${img} || true`);
+        cy.dropFile(img, img.split('/').pop(), '.pf-c-file-upload');
+        cy.get('.pf-c-alert__title')
+          .contains('File type extension')
+          .should('not.exist');
+      }
+    });
+
+    it('ID(CNV-4891) It shows a warning message when image format is not supported', () => {
+      pvc.form.open();
+      cy.exec(`touch ${invalidImage} || true`);
+      cy.dropFile(invalidImage, invalidImage.split('/').pop(), '.pf-c-file-upload');
+      cy.contains('File type extension').should('be.visible');
+    });
+
+    it('ID(CNV-5176) It shows an error when uploading data to golden OS again', () => {
+      cy.createDataVolume(TEMPLATE_BASE_IMAGE, OS_IMAGES_NS);
+      pvc.form.open();
+      pvc.form.selectOS(os);
+      cy.get('.pf-c-alert__title')
+        .contains('Operating system source already defined')
+        .should('exist');
+    });
+
+    it('ID(CNV-5041) VM can be up after deleting the uploaded PVC', () => {
+      vm.create(vmData);
+      vm.stop(vmData);
+      // only delete template pvc for ocs, hpp does not support this
+      if (Cypress.env('STORAGE_CLASS') === 'ocs-storagecluster-ceph-rbd') {
+        cy.deleteResource({
+          kind: 'DataVolume',
+          metadata: {
+            name: TEMPLATE_BASE_IMAGE,
+            namespace: OS_IMAGES_NS,
+          },
+        });
+      }
+      vm.start(vmData);
+      vm.delete();
+    });
+  });
+
+  describe('test pvc upload via cli', () => {
+    it('ID(CNV-5044) Verify boot source is available for template after upload via cli', () => {
+      cy.exec(
+        `test -f ${Cypress.env(
+          'UPLOAD_IMG',
+        )} || curl --fail -L ${ProvisionSource.URL.getSource()} -o ${Cypress.env('UPLOAD_IMG')}`,
+        { timeout: 600000 },
+      );
+      if (Cypress.env('STORAGE_CLASS') === 'ocs-storagecluster-ceph-rbd') {
+        cy.exec(
+          `virtctl image-upload dv ${TEMPLATE_BASE_IMAGE} --image-path=${Cypress.env(
+            'UPLOAD_IMG',
+          )} --size=1Gi --storage-class=ocs-storagecluster-ceph-rbd --access-mode=ReadWriteMany --block-volume -n ${OS_IMAGES_NS} --insecure || true`,
+        );
+      }
+      if (Cypress.env('STORAGE_CLASS') === 'hostpath-provisioner') {
+        cy.exec(
+          `virtctl image-upload dv ${TEMPLATE_BASE_IMAGE} --image-path=${Cypress.env(
+            'UPLOAD_IMG',
+          )} --size=1Gi --storage-class=hostpath-provisioner --access-mode=ReadWriteOnce -n ${OS_IMAGES_NS} --insecure || true`,
+        );
+      }
```

Comment:
Is this something we should consider extracting out into a utility function?

Reply:
move it to support/index